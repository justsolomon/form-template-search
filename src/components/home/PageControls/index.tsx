import Button from 'components/global/Button';
import { ReactComponent as ArrowIcon } from 'assets/vectors/arrow-right.svg';
import { useAppDispatch, useAppSelector } from 'redux/hooks';
import { goToPage } from 'redux/slices/templateSlice';
import styles from './pageControls.module.scss';

const PageControls = () => {
  const { currentPage, prevPage, nextPage, numOfPages } = useAppSelector(
    (store) => store.template.pagination,
  );
  const dispatch = useAppDispatch();

  return numOfPages ? (
    <div className={styles['page-controls']}>
      <Button
        variant="transparent"
        leftIcon={prevPage ? <ArrowIcon /> : null}
        onClick={() => {
          dispatch(goToPage('prev'));
        }}
        className={styles['page-controls__prev-button']}
      >
        Previous
      </Button>

      <div className={styles['page-controls__page-info']}>
        <span className={styles['page-controls__current-page']}>
          {currentPage}
        </span>{' '}
        of {numOfPages}
      </div>

      <Button
        variant="transparent"
        rightIcon={nextPage ? <ArrowIcon /> : null}
        onClick={() => {
          dispatch(goToPage('next'));
        }}
        className={styles['page-controls__next-button']}
      >
        Next
      </Button>
    </div>
  ) : null;
};

export default PageControls;
