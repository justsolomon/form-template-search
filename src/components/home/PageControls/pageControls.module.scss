@import 'styles/variables';
@import 'styles/mixins';

.page-controls {
  @include flex-center();

  margin: 75px 50px 0 50px;
  justify-content: space-between;

  button svg {
    width: 6px;
    height: 12px;
  }

  &__prev-button {
    svg {
      transform: rotate(180deg);
    }
  }

  &__current-page {
    height: 30px;
    width: 30px;
    margin-right: 8px;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    border: 1px solid $primary-color;
    border-radius: $border-radius--md;
  }

  @media #{$media-mobile} {
    margin: 60px 12px 0 12px;
  }

  @media #{$media-mobile--sm} {
    margin: 50px 0 0 0;

    button {
      font-size: $font-size--base;

      svg {
        width: 5px;
        height: 9px;
      }
    }
  }
}
