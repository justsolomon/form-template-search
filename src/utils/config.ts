export const baseUrl =
  'https://front-end-task-dot-result-analytics-dot-fpls-dev.uc.r.appspot.com/api/v1';
