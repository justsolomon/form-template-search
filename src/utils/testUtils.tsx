import {
  queries,
  Queries,
  render as rtlRender,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import rootReducer from 'redux/rootReducer';
import { Store } from 'redux';

interface CustomRenderOptions extends RenderOptions {
  preloadedState: Record<string, any>;
  store: Store;
}

const render = <
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
>(
  ui: React.ReactElement,
  {
    preloadedState,
    store = configureStore({ reducer: rootReducer, preloadedState }),
    ...renderOptions
  }: CustomRenderOptions,
): RenderResult => {
  function Wrapper({ children }: { children: React.ReactNode }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return rtlRender(ui, {
    wrapper: Wrapper as React.ComponentType,
    ...renderOptions,
  });
};

export * from '@testing-library/react';
export { render };
