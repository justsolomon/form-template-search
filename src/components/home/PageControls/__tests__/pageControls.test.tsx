import store from 'redux/store';
import { render } from 'utils/testUtils';
import PageControls from '..';

describe('<PageControls />', () => {
  it('should render the PageControls component correctly', () => {
    const { asFragment } = render(<PageControls />, {
      preloadedState: {},
      store,
    });

    expect(asFragment).toMatchSnapshot();
  });
});
