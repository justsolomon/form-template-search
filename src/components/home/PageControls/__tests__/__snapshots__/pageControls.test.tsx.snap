// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageControls /> should render the PageControls component correctly 1`] = `[Function]`;
